### ISSUE TITLE
Bug: [member-ordering] function class fields with default value are identified as class methods

### ISSUE THREAD

**Issue by tao-cumplido** [typescript-eslint/typescript-eslint#5187]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYGwhgzhAEAqCmEAu0DeBYAUNaYBc0AFAJTQC8AfNAG4D2AlgCblGmVoC+A3FjgEYB+AiXJU6THpg5A&eslintrc=N4KABGBEAOCGBOBnApvSAuKABALgT2mUQGN4BLaHAWiIBsyA7HAejiVUgBpwp4BXWkQxhQECJFwEipCtTqMWAW2SKARqioB7eABNUjAObCA2jzFRU8bZDNgAujwC+IR0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A

### Repro Code

```typescript
class Test {
  a: () => void = () => {};
  b?: () => void;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/member-ordering": ["error"]
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

No error.

### Actual Result

```
Member b should be declared before all public instance method definitions. 3:3 - 3:17
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.28.0` |
| `@typescript-eslint/parser`            | `5.28.0` |
| `TypeScript`                           | `4.7.2` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `16.15.0` |

**Comment by bradzacher** [comment]
This is intentional.
This was added to the rule 5 years ago (https://github.com/bradzacher/eslint-plugin-typescript/pull/46).

Class properties that declare function values are thought of by most to be methods.
Particularly when using an arrow function, they're thought of as "auto bound methods".

**Comment by tao-cumplido** [comment]
> Class properties that declare function values are thought of by most to be methods.
I must admit I find that hard to believe. Why would someone use class fields with function values like methods? I specifically use function values for class fields as callbacks that can be set like other properties. Why should there be a distinction for fields with and without a default value?

I know it would add complexity but couldn't it be made configurable at least, with the current way being the default? I wouldn't mind working on it myself.

**Comment by Josh-Cena** [comment]
Related: https://github.com/mdn/content/issues/11237

I echo with the sentiment that class fields that happen to be functions have very different semantics from methods (for one important difference, fields are on instances, not instance prototypes). The fact that an initializer makes it a method also doesn't seem right, since fields are mutable.

**Comment by bradzacher** [comment]
Why is there a distinction between with value and without? Because in those cases intent is important.
Not having a value signals the property's value will be changed later - by definition it **must** be changed later.

That doesn't preclude props with a value from being reassigned ofc, but to be fair - neither does a true method definition (yes, TS treats methods as writable and there is no way to make them readonly).
But the intent and general case is that a prop with an arrow function value won't be reassigned later because it is being used as an auto-bound method.

In an ideal world you would require the `readonly` keyword for a prop to be treated as a method. However in the broader TS community it's very rare that people code like this - people really have an aversion to making properties `readonly`.

So IMO it does make complete sense that value-less props are "fields" and props with fn values are "methods".

---

Investigating further back to the tslint rule upon which our rule was originally based and then evolved in parallel to - I can see evidence of it working this way even further back (6 years ago).

---

Considering it was intentionally changed to specifically be this way 5 years ago, and this is the first time someone has asked for it to not work this way reinforces the idea that most people think of arrow function props as methods.

Therefore making it configurable is not a change that benefits the wider community, so it us not worth us maintaining in our rule.

You're free to fork the rule if you'd like to change this behaviour for your project!

**Comment by tao-cumplido** [comment]
Ok, thanks for the detailed explanation.
